core/WFCORE-1427_ Convenient_declaration_of_server_suspend_timeout.adoc

```diff
@@ -0,0 +1,235 @@
+** Meaning of the suspend-timeout should be the same as the 'timeout' param currently available for the 'shutdown' operation. A value of less than 0 means wait indefinitely, a value of 0 means don't wait, a value larger than zero means the server will wait up to this many seconds for all active requests to finish. Default value is 0. Value is in seconds.
+** Operations like 'shutdown' that already have a 'timeout' param are excepted from this requirement; no need to add a semantically equivalent 'suspend-timeout'.
+* Relevant operations include 'reload' for a server or HC and 'restart-servers' on the domain root resource and on the server group resource. An audit should be performed to check for others as well.
+* All high level CLI commands that result in a server stopping handling of requests should also accept a 'suspend-timeout' parameter.
+* For a Host Controller 'reload' operation or high level CLI command, the suspend-timeout applies to the servers, not to the HC process. HC suspend is not supported. Any 'suspend-timeout' is irrelevant if the 'restart-servers' param is set to 'false' since in that case no server will stop normal handling of requests.
+* The 'suspend-timeout' is a per server, not an aggregate overall timeout for operations that affect multiple servers.
+* The 'suspend-timeout' is only for a server suspending before stopping normal request handling, not for the time take for any other aspect of the operation, e.g. the time to restart/reload once the server is suspended.
+
+=== Nice-to-Have Requirements
+* HAL support. This should be a hard requirement for a subsequent release but is a nice to have initially.
+* Reject the use of the 'suspend-timeout' param for an HC reload if the 'restart-servers' param is set to 'false'. If servers are not being restarted there is not 'suspend' involved, so the combination is non-sensical.
+** If this is done it should be done consistently in low level op handling on the server side and in CLI high level command processing  on the client side. IOW the high level command should not accept parameter combinations that are then rejected on the server.
+* Parallelization of multi-server suspend where not already done and where feasible and not an incompatible behavior change.
+** This requirement is not meant to imply that the suspend part of task execution should be separated from other parts (e.g. in a reload case, suspend all servers in a parallel and the proceed on to the rest of reloading the servers).
+
+=== Non-Requirements
+* For an operation affecting multiple servers, treating the 'suspend-timeout' as being some overall timeout. The timeout applies to each server.
+* Cancellation of the operations using the :resume operation. For example, cancellation of a restart meanwhile the servers are waiting for the suspend-timeout.
+
+== Design Notes
+
+The following list shows the operations that will use the suspend-timeout attribute:
+
+[source]
+--
+[domain@localhost:9990 /] /host=*:reload
```

Comment:
Yes, you're right, looks like this operation is also a candidate to include a suspend timeout. Keep in mind this operation is not using yet an attribute called timeout, but the high-level `shutdown` operation includes one, although it is not used in domain mode, see:

```
[domain@localhost:9990 /] help shutdown
```

Reply:
Resolving this conversation, the above comments are now obsolete because we finally are not going to add a new attribute to the `shutdown` operation, instead we will be able to use  first a `/host=*:suspend-servers(suspend-timeout=*)` to suspend the servers and later, execute the shutdown.